tests/multi_threaded.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+mod common;
+
+use common::{CounterInnerMutSubscriber, CounterSubscriber, CounterSubscriberWithData};
+use event_manager::{EventManager, EventSubscriber, MutEventSubscriber, SubscriberOps};
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+// Showcase how you can manage subscribers of different types using the same event manager
+// in a multithreaded context.
+#[test]
+fn test_diff_type_subscribers() {
+    let mut event_manager = EventManager::<Arc<Mutex<dyn MutEventSubscriber + Send>>>::new()
+        .expect("Cannot create event manager.");
+
+    // The `CounterSubscriberWithData` expects to receive a number as a parameter that represents
+    // the number it can use as its inner Events data.
+    let data_subscriber = Arc::new(Mutex::new(CounterSubscriberWithData::new(1)));
+    data_subscriber.lock().unwrap().trigger_all_counters();
+
+    let counter_subscriber = Arc::new(Mutex::new(CounterSubscriber::new()));
+    counter_subscriber.lock().unwrap().trigger_event();
+
+    // Saving the ids allows to modify the subscribers in the event manager context (i.e. removing
+    // the subscribers from the event manager loop).
+    let ds_id = event_manager.add_subscriber(data_subscriber);
+    let cs_id = event_manager.add_subscriber(counter_subscriber);
+
+    let thread_handle = thread::spawn(move || {
+        // We cannot run the event manager loop for ever. Let's just run it for 100 iterations.
```

Comment:
Yes we can but we choose not to :smile: I'd rephrase to say that 100 runs are *enough* because we don't want an infinite test